### DISCUSSION TITLE
Enable reverse proxy on multiple containers with single host listening to same port

### DISCUSSION THREAD

**Discussion by shetesagar87** [nginx/nginx#315] [Q&A] [answered]
Hello All, 

Below is the thought which I wanted to implement and facing issues while achieving it. 

1) Container 1 - nginx
2) Container 2 - apache

Have a reverse proxy on top of it. So that all the traffic is routed via the reverse proxy only. We should list on single port and route the traffic based on /nginx or /apache based on request URL. 

`
server {
    listen 8080;
 
    location /nginx {
        proxy_pass         http://docker-nginx;
        proxy_redirect     off;
       
    }

     location /apache {
        proxy_pass       /////  http://docker-apache;
        proxy_redirect     off;
    }
}
`

Is this even possible ? Do share your insights on this.

**Comment by viktorberezikov** [comment, marked as answer]
In your example, requests for the endpoint `/nginx` are routed to the upstream `http://docker-nginx`. And requests for the endpoint `/apache` are routed to the upstream `http://docker-apache`. The important point is that upstream itself should return links that include its endpoint. Or set up `proxy_redirect`. This enables the next requests from the client/browser to be routed correctly.
